### PR TITLE
[OP-1708, OP-1709] Add patch types: helmDepUpdate & yaml

### PR DESCRIPTION
**Pull request by applejag**
OP-1708, OP-1709

- Updated Go to 1.22.2

- Refactored away from line-based patching, as only the Regex patches uses that

- Moved `file` field into regex **(breaking change!)**

  ```yaml
  # before:
  - url: tmp/upstream-test
    patches:
      - file: go.mod   # <---
        regex:
          match: '(github.com/joho/godotenv) v.*'
          replace: '{{ index .Groups 1 }} {{ .Version }}'

  # after:
  - url: tmp/upstream-test
    patches:
      - regex:
          file: go.mod   # <---
          match: '(github.com/joho/godotenv) v.*'
          replace: '{{ index .Groups 1 }} {{ .Version }}'
  ```

- Added YAML patch via VMWare's yaml-jsonpath: <https://pkg.go.dev/github.com/vmware-labs/yaml-jsonpath/pkg/yamlpath>

- Added Helm dep update patch by just running `helm dep update` in the given directory

- Added Helm & Git LFS to Dockerfile

- Added config page where you can test out package configs

  image

### CODE REVIEW

File: pkg/patch/patch.go [hunk 1]
```diff
@@ -47,85 +45,87 @@ type TemplateContextRegex struct {
...
-func patchSingleLine(patch config.PackageRepoPatch, tmplCtx TemplateContext, line []byte) ([]byte, error) {
-	switch {
-	case patch.Regex != nil:
-		return patchSingleLineRegex(*patch.Regex, tmplCtx, line)
-	default:
-		return nil, errors.New("missing patch type config")
-	}
-}
+		fullMatchStart := groupIndices[0]
+		fullMatchEnd := groupIndices[1]
```
Comment (jadolg) [inline pkg/patch/patch.go:110]: Are these indices guaranteed to exist?

File: pkg/patch/filestore.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+// more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program.  If not, see <http://www.gnu.org/licenses/>.
+
```
[the remark this answers is not in the document]
Comment (applejag) [reply pkg/patch/filestore.go:17]: Yea sure, I'll add some comments.

In short, it's for making the code testable

File: pkg/patch/patch.go (hunk 1 above)
Comment (applejag) [reply pkg/patch/patch.go:110]: Yes.

- `groupIndices[0]`: start index of full match
- `groupIndices[1]`: end index of full match
- `groupIndices[2]`: start index of group 1
- `groupIndices[3]`: end index of group 1
- `groupIndices[4]`: start index of group 2
- `groupIndices[5]`: end index of group 2

Only the first two are guaranteed to exist. The other depend on how many groups you define in your regex.

The Go regex API sucks. Just tossing around some structs (or even just some arrays) in here would improve it by a landslide.

File: pkg/config/yamlpath.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+type YAMLPathPattern struct {
+	YAMLPath *yamlpath.Path
+	Source   string
+}
+
+// Ensure the type implements the interfaces
+var _ pflag.Value = &YAMLPathPattern{}
+var _ encoding.TextUnmarshaler = &YAMLPathPattern{}
+var _ jsonSchemaInterface = YAMLPathPattern{}
```
[the remark this answers is not in the document]
Comment (applejag) [reply pkg/config/yamlpath.go:37]: I like this pattern as you then have the compile error close to the source code of the type.

And then as for interfaces like `encoding.TextUnmarshaler` they're looked up via reflection during runtime inside the YAML and JSON libraries. So there it plays a bigger role.

**Review by jadolg — APPROVED** [review]

**Pull request merged**